files/unarchive.py

```diff
@@ -347,7 +347,7 @@ def is_unarchived(self):
             version = pcs[1]
             ostype = pcs[2]
             size = int(pcs[3])
-            path = pcs[7]
+            path = unicode(pcs[7], 'utf-8')
```

Comment:
For most modules there's two sets of tests wrt python3 and modules -- there's a syntax test which attempts to compile the source into bytecode using python3 and integration tests.  The compile test won't pick this up because ```unicode()``` could be a user defined function.  The integration tests only exercise the code paths needed for the integration test to pass and also are only running on the modules which already pass (to help catch regressions).  Since unarchive doesn't pass yet, the integration tests aren't being run on it.

Here's the blacklist of integration tags: https://github.com/ansible/ansible/blob/devel/test/utils/shippable/python3-test-tag-blacklist.txt